alerter/TemperatureChangeNotifier.java

```diff
@@ -0,0 +1,27 @@
+public class TemperatureChangeNotifier {
+
+    static int alertInCelcius(float fahrenheit) {
+        int alertFailureCount = 0;
+        float celcius = (fahrenheit - 32) * 5 / 9;
+        TemperatureThresholdManipulatorImpl temperatureThresholdManipulatorImpl = new TemperatureThresholdManipulatorImpl();
```

Comment:
defining an interface is to abstract production code from knowing who the implementor is. Which means that in test envt, this interface is implemented by test code and in deployment, it is actual production component.

But here the production code knows the class and instantiates the impln object. the interface is not used.